pipeline_steps/gate.groovy

```diff
@@ -0,0 +1,194 @@
+void testPullRequestOnly(String repo_name){
+    List jobNames = Hudson.instance.getAllItems(org.jenkinsci.plugins.workflow.job.WorkflowJob)*.fullName
+
+    def parallelBuilds = [:]
+
+    // Cannot do for (job in jobNames), see:
+    // https://jenkins.io/doc/pipeline/examples/#parallel-multiple-nodes
+    for (j in jobNames) {
+        def job = j
+        def m = job =~ /GATE_${repo_name}/
+        if (m) {
+            parallelBuilds[job] = {
+                build(
+                    job: job,
+                    wait: true,
+                    parameters: [
+                        [
+                            $class: "StringParameterValue",
+                            name: "RPC_GATING_BRANCH",
+                            value: RPC_GATING_BRANCH,
+                        ],
+                        [
+                            $class: "StringParameterValue",
+                            name: "BRANCH",
+                            value: sha1,
+                        ]
+                    ]
+                )
+            }
+        }
+    }
+
+    parallel parallelBuilds
+
+    build(
+      job: "Merge-Pull-Request",
+      wait: false,
+      parameters: [
+        [
+          $class: "StringParameterValue",
+          name: "RPC_GATING_BRANCH",
+          value: RPC_GATING_BRANCH,
+        ],
+        [
+          $class: "StringParameterValue",
+          name: "pr_repo",
+          value: ghprbGhRepository,
+        ],
+        [
+          $class: "StringParameterValue",
+          name: "pr_number",
+          value: ghprbPullId,
+        ],
+        [
+          $class: "StringParameterValue",
+          name: "commit",
+          value: ghprbActualCommit,
+        ],
+      ]
+    )
+}
+
+
+void testPullRequestWithAntecedents(String repo_name){
```

Comment:
Comment with a high level overview of the logic of this function would be useful here.